sky/clouds/gcp.py

```diff
@@ -742,7 +742,8 @@ def check_credentials(cls) -> Tuple[bool, Optional[str]]:
         credentials, project = google.auth.default()
         crm = googleapiclient.discovery.build('cloudresourcemanager',
                                               'v1',
-                                              credentials=credentials)
+                                              credentials=credentials,
+                                              cache_discovery=False)
```

Comment:
Should we benchmark the performance degrade and make sure it is tolerable? I think it should be marginal though

Reply:
Yep, the performance is similar. Tested with:
```
multitime -n 5 python -c "from sky.adaptors import gcp; gcp.build('cloudresourcemanager', 'v1', credentials=None, cache_discovery=False); gcp.build('cloudresourcemanager', 'v1', credentials=None, cache_discovery=False)"
===> multitime results
1: python -c "from sky.adaptors import gcp; gcp.build('cloudresourcemanager', 'v1', credentials=None, cache_discovery=False); gcp.build('cloudresourcemanager', 'v1', credentials=None, cache_discovery=False)"
            Mean        Std.Dev.    Min         Median      Max
real        1.174       0.009       1.165       1.174       1.188       
user        1.210       0.025       1.170       1.214       1.245       
sys         0.243       0.021       0.220       0.238       0.278
```

```
multitime -n 5 python -c "from sky.adaptors import gcp; gcp.build('cloudresourcemanager', 'v1', credentials=None); gcp.build('cloudresourcemanager', 'v1', credentials=None)"
INFO:googleapiclient.discovery_cache:file_cache is only supported with oauth2client<4.0.0
INFO:googleapiclient.discovery_cache:file_cache is only supported with oauth2client<4.0.0
INFO:googleapiclient.discovery_cache:file_cache is only supported with oauth2client<4.0.0
INFO:googleapiclient.discovery_cache:file_cache is only supported with oauth2client<4.0.0
INFO:googleapiclient.discovery_cache:file_cache is only supported with oauth2client<4.0.0
===> multitime results
1: python -c "from sky.adaptors import gcp; gcp.build('cloudresourcemanager', 'v1', credentials=None); gcp.build('cloudresourcemanager', 'v1', credentials=None)"
            Mean        Std.Dev.    Min         Median      Max
real        1.241       0.028       1.211       1.230       1.293       
user        1.262       0.038       1.220       1.269       1.323       
sys         0.262       0.019       0.237       0.258       0.294 
```